modin/config/envvars.py

```diff
@@ -67,6 +74,81 @@ def get_help(cls) -> str:
         return help
 
 
+class EnvWithSibilings(
+    EnvironmentVariable,
+    # we have to pass anything here in order to derive from 'EnvironmentVariable',
+    # this doesn't force child classes to have 'str' type, they actually can be any type
+    type=str,
+):
+    """Ensure values synchronization between sibling parameters."""
+
+    _update_sibling = True
+
+    @classmethod
+    def _sibling(cls) -> type["EnvWithSibilings"]:
+        """Return a sibling parameter."""
+        raise NotImplementedError()
+
+    @classmethod
+    def get(cls) -> Any:
+        """
+        Get parameter's value and ensure that it's equal to the sibling's value.
+
+        Returns
+        -------
+        Any
+        """
+        if cls._sibling()._value is _UNSET and cls._value is _UNSET:
+            old_v: Any
+            new_v: Any
+            try:
+                old_v = cls._sibling()._get_raw_from_config()
+            except KeyError:
+                old_v = _UNSET
+            try:
+                new_v = cls._get_raw_from_config()
+            except KeyError:
+                new_v = _UNSET
+            if old_v is not _UNSET and new_v is _UNSET:
+                if not _TYPE_PARAMS[cls.type].verify(old_v):
+                    raise ValueError(f"Unsupported raw value: {old_v}")
+                old_v = _TYPE_PARAMS[cls.type].decode(old_v)
+                cls._sibling()._value = old_v
+                cls._sibling()._value_source = ValueSource.GOT_FROM_CFG_SOURCE
+
+                cls._value = old_v
```

Comment:
```suggestion
                cls._value = new_v
```
?

Reply:
no, this code is located under the following condition:
```python
if old_v is not _UNSET and new_v is _UNSET:
```
meaning, that the `old_v` carries a valid value here